[RuijianLu] iP

RuijianLu opened the pull request:
# Ruijian iP Pull Request

**LuToDo** is edited from _Duke_ and have basic features of a task list managing app.

Features added:
- Enable basic operations to the task list
- Enable parsing some basic commands
- Enable saving the list to a txt file and load it

Structural changes:
1. `Package` all `classes`
2. Divide the codes to proper classes
3. Set up gradle

This is the main class of the project codes.
```java
public class LuToDo {

}
```
Things to do later:

- [ ] Follow requirements of the course website.

I hope this tasklist manager can give you good experience.👍 

> This is a quote.

link to the github repo: [https://github.com/RuijianLu/ip](url)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

RuijianLu committed:
Initialize the bot with static greeting and exiting.

RuijianLu committed:
Enable echo, but badly structured

RuijianLu committed:
Enable add to list and restructure.

RuijianLu committed:
Enable mark as done and unmark. Also restructured.

RuijianLu committed:
Enable adding different types of tasks, but polymorphism is not implemented now

RuijianLu committed:
try to set autotest

RuijianLu committed:
create Exception file

RuijianLu committed:
Enable some exception handling, although badly structured.

RuijianLu committed:
Enable deleting tasks

RuijianLu committed:
Enable creating task history file

RuijianLu committed:
Enable read file and make list

RuijianLu committed:
Enable tasklist to file

RuijianLu committed:
Handle when cannot read file

RuijianLu committed:
Merge branch 'branch-Level-7'

RuijianLu committed:
Fix a little bug about deleting and add tasklist file to gitignore

RuijianLu committed:
Enable understanding ISO format date and time

RuijianLu committed:
Merge branch 'branch-Level-8'

RuijianLu committed:
Add logfile to gitignore

RuijianLu committed:
Add Ui.java, TaskList.java, Storage.java, Parser.java

RuijianLu committed:
Add multiple classess to do more OOP

RuijianLu committed:
Add more OOP and make the program work

RuijianLu committed:
Merge branch 'branch-A-MoreOOP'

RuijianLu committed:
Bring all java files into packages

RuijianLu committed:
Merge remote-tracking branch 'origin/add-gradle-support'

# Conflicts:
#	text-ui-test/runtest.sh

RuijianLu committed:
Delete not used import statements

RuijianLu committed:
Delete codes not needed

RuijianLu committed:
Setup gradle

RuijianLu committed:
Add test to parse method

RuijianLu committed:
Add test to splitTaskInfo method

RuijianLu committed:
Fix a bug of splitTaskInfo method

RuijianLu committed:
Ready for first publish

RuijianLu committed:
Change file dir to be ready for first release

RuijianLu committed:
Create the very first JavaDoc

RuijianLu committed:
Add javadoc to almost all the public classes and methods.

RuijianLu committed:
Add javadoc to main class

RuijianLu committed:
Fulfil conding standard

RuijianLu committed:
Enable search tasks

RuijianLu committed:
Merge branch 'branch-A-JavaDoc'

RuijianLu committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/lutodo/commands/ShowListCommand.java

RuijianLu committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/lutodo/tasks/Task.java

saajidshaik02 commented on .gitignore:
Good you added file text to gitignore!

saajidshaik02 commented on src/main/java/lutodo/commands/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+                        + "\nNow you have " + tasks.size() + " tasks in the list.\n");
+            } catch (IndexOutOfBoundsException e) {
+                System.out.print("""
+                        The description of the deadline task must include the ddl time, please try again.
```
Consider adding string constants at the start of the code, then refer for readability!

saajidshaik02 commented on src/main/java/lutodo/commands/Command.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * Judges if two commands can be considered same, mainly used in tests.
+     *
+     * @param otherCommand The other command to be compared.
```
Good documentation of functions!

saajidshaik02 commented on src/main/java/lutodo/parser/Parser.java:
```diff
@@ -0,0 +1,82 @@
+            return new UnknownCommand(fullCommand);
+        } else {
+            String taskType = splitTaskInfo(fullCommand)[0];
+            if(taskType.equalsIgnoreCase("bye")) {
```
"if(" needs 1 extra spacing

saajidshaik02 commented on src/main/java/lutodo/storage/Storage.java:
```diff
@@ -0,0 +1,97 @@
+                    default:
+                        System.out.println("Cannot read: " + taskMessage);
+                }
+                if(taskMessage.charAt(4) == 'X') {
```
+1 spacing

saajidshaik02 commented on src/main/java/lutodo/tasks/EventTask.java:
```diff
@@ -0,0 +1,52 @@
+     *
+     * @param wholeInfo Information of the task, including description, start time and end time.
+     */
+    public EventTask(String wholeInfo) {
```
maybe can document what the 0 1 and 2 means?

saajidshaik02 reviewed (commented):
LGTM. Just need format code abit more. Jiayou!

RuijianLu committed:
Add GUI to the chatbot

RuijianLu committed:
Merge branch 'branch-Level-10'

RuijianLu committed:
Correct coding standard errors

RuijianLu committed:
Modify build.gradle for correct jar file

RuijianLu committed:
Add most simple assertions

RuijianLu committed:
Fix a major bug

Fixed the bug about the behavior when user input lacks information.
edited commands:
* delete
* find
* mark
* unmark

RuijianLu committed:
Add assertions

Add assertions to check the task message is successfully conveyed.
Modified commands:
* find
* delete
* mark

RuijianLu committed:
Merge pull request #1 from RuijianLu/branch-A-Assertions

Add assertions and fixed a bug

RuijianLu committed:
Merge branch 'master' of https://github.com/RuijianLu/ip

RuijianLu committed:
Merge branch 'master' into branch-A-CodeQuality

RuijianLu committed:
Improve code quality

Make the method shorter and make the code more SLAP.
modified files:
* AddCommand.java
* DeleteCommand.java
* FindCommand.java
* MarkCommand.java

RuijianLu committed:
Merge pull request #2 from RuijianLu/branch-A-CodeQuality

Branch A code quality

RuijianLu committed:
Merge branch 'master' of https://github.com/RuijianLu/ip

RuijianLu committed:
Enable searching deadline date and event start time

Add a isMatching method to manage the search and can be improved later.

RuijianLu committed:
Merge branch 'branch-BCD-Extension'

RuijianLu committed:
Add Ui.png

RuijianLu committed:
Modify head commands by code standard

RuijianLu committed:
A-UserGuide

RuijianLu committed:
Revert "A-UserGuide"

This reverts commit 95cff66884294c42b1fbb9535bbd030a77772c2e.

RuijianLu committed:
Modify README.md

RuijianLu committed:
Modify README.md for better visual

RuijianLu committed:
Modify 8 javadocs for coding standard

RuijianLu committed:
Add javadoc to fuction main

RuijianLu committed:
Rename Ui.png.png to Ui.png

RuijianLu committed:
Modify build.gradle to change version of JAR to v1.0
